src/layers/learning/gru.cpp

```diff
@@ -207,7 +207,8 @@ void gru_layer<TensorDataType, Layout, Device>::setup_gpu() {
   // RNN descriptor
   size_t dropout_state_size;
   CHECK_CUDNN(cudnnDropoutGetStatesSize(handle, &dropout_state_size));
-  cudnn::DropoutDescriptor dropout_desc(0.f, nullptr, dropout_state_size, 0);
+  cudnn::DropoutDescriptor dropout_desc;
+  dropout_desc.set(0.f, nullptr, dropout_state_size, 0);
```

Comment:
On the one hand, that's fair that this will only be available in cuDNN-land for now, but my understanding is that this code will still need to run on HIP-based machines in the future. Is there harm in doing the modifications that we can do now?

Reply:
Most of the changes would be overwritten in the other PR since we're using the new RNN API in cuDNN 8.

Reply:
Ah, got it. Will revert prior to merge.

Reply:
Done.